x-pack/plugins/session_view/public/components/SessionView/index.tsx

```diff
@@ -89,22 +102,89 @@ export const SessionView = ({ sessionEntityId, height }: SessionViewDeps) => {
     );
   };
 
+  const toggleDetailPanel = () => {
+    setIsDetailMounted(!isDetailMounted);
+    if (!isDetailOpen) setIsDetailOpen(true);
+  };
+
   if (!data.length) {
     return renderNoData();
   }
-
+console.log(selectedProcess)
   return (
     <>
-      <EuiSearchBar query={searchQuery} onChange={onSearch} />
-      <div css={styles.processTree}>
-        <ProcessTree
-          sessionEntityId={sessionEntityId}
-          forward={data}
-          searchQuery={searchQuery}
-          selectedProcess={selectedProcess}
-          onProcessSelected={onProcessSelected}
-        />
-      </div>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiSearchBar query={searchQuery} onChange={onSearch} />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButton onClick={toggleDetailPanel} iconType="list" fill>
+            Detail Panel
+          </EuiButton>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiSplitPanel.Outer direction="row" color="transparent" borderRadius="none" css={styles.outerPanel}>
+        <EuiSplitPanel.Inner paddingSize="none" css={styles.treePanel}>
+          <div css={styles.processTree}>
+            <ProcessTree
+              sessionEntityId={sessionEntityId}
+              forward={data}
+              searchQuery={searchQuery}
+              selectedProcess={selectedProcess}
+              onProcessSelected={onProcessSelected}
+            />
+          </div>
+        </EuiSplitPanel.Inner>
+        {isDetailOpen && (
+          <EuiSplitPanel.Inner
+            paddingSize="s"
+            color="plain"
+            css={isDetailMounted ? styles.detailPanelIn : styles.detailPanelOut}
+            onAnimationEnd={() => {
+              if (!isDetailMounted) setIsDetailOpen(false);
+            }}
+          >
+            {selectedProcess && (
+              <>
+                <EuiTitle size="xs"><span>Command Detail</span></EuiTitle>
+                <EuiSpacer />
+                {selectedProcess.events.map((event) => {
+                  return (<EuiDescriptionList
+                    type="column"
+                    compressed
+                    listItems={Object.keys(event).map((title) => ({
+                      title,
+                      description: JSON.stringify(event[title as keyof ProcessEvent], null, 4),
+                    }))}
```

Comment:
Can we precompute these values outside of the returned JSX, it's a bit hard to read when it's all inlined

Reply:
yea this is just for place holding, will replace it in the next commit